EpsilonCCS/Src/CanParsers/MotorCanParser.c

```diff
@@ -0,0 +1,252 @@
+            motor1DetailsData.railSupply3_3v = arrayToFloat(data[4]);
+            break;
+
+        case M1_HEATSINK_TEMPERATURE_CAN_ID:
+            motor1DetailsData.motorTemperature = arrayToFloat(data[0]);
+            motor1DetailsData.heatSinkTemperature = arrayToFloat(data[4]);
+            break;
+
+        case M1_DSP_BOARD_TEMPERATURE_CAN_ID:
+            motor1DetailsData.dspBoardTemp = arrayToFloat(data[0]);
+            break;
+
+        case M1_ODOMETER_BUS_AMPHOURS_CAN_ID:
+            motor1DetailsData.odometer = arrayToFloat(data[0]);
+            motor1DetailsData.dcBusAmpHours = arrayToFloat(data[4]);
+            break;
+
+        case M1_SLIP_SPEED_CAN_ID:
+            motor1DetailsData.slipSpeed = arrayToFloat(data[4]);
+            break;
+    }
+}
+
+void parseMotorFaultMessage(uint8_t motorId, uint8_t* data)
+{
+    struct MotorErrorFlags* errorData;
+    struct MotorLimitFlags* limitData;
+    unsigned char* rxErrorData;
+    unsigned char* txErrorData;
+
+    if (motorId == M0_ID)
+    {
+        errorData = &motorFaultsData.m0ErrorFlags;
+        limitData = &motorFaultsData.m0LimitFlags;
+        rxErrorData = &motorFaultsData.m0CanRxErrorCount;
+        txErrorData = &motorFaultsData.m0CanTxErrorCount;
+    }
+    else if (motorId == M1_ID)
+    {
+        errorData = &motorFaultsData.m1ErrorFlags;
+        limitData = &motorFaultsData.m1LimitFlags;
+        rxErrorData = &motorFaultsData.m1CanRxErrorCount;
+        txErrorData = &motorFaultsData.m1CanTxErrorCount;
+    }
+    else
+    {
+        return;
+    }
+
+    (*limitData).outputVoltagePwm =
+        data[0] & OUTPUT_VOLTAGE_PWM_LIMIT_MASK;
+    (*limitData).motorCurrent =
+        data[0] & MOTOR_CURRENT_LIMIT_MASK;
+    (*limitData).velocity =
+        data[0] & VELOCITY_LIMIT_MASK;
+    (*limitData).busCurrent =
+        data[0] & BUS_CURRENT_LIMIT_MASK;
+    (*limitData).busVoltageUpper =
+        data[0] & BUS_VOLTAGE_UPPER_LIMIT_MASK;
+    (*limitData).busVoltageLower =
+        data[0] & BUS_VOLTAGE_LOWER_LIMIT_MASK;
+    (*limitData).ipmOrMotorTemperature =
+        data[0] & IPM_OR_MOTOR_TEMPERATURE_LIMIT_MASK;
+
+    (*errorData).motorOverSpeed =
+        data[1] & MOTOR_OVER_SPEED_MASK;
+    (*errorData).softwareOverCurrent =
+        data[0] & SOFTWARE_OVER_CURRENT_MASK;
+    (*errorData).dcBusOverVoltage =
+        data[0] & DC_BUS_OVER_VOLTAGE_MASK;
+    (*errorData).badMotorPositionHallSequence =
+        data[0] & BAD_MOTOR_POSITION_HALL_SEQUENCE_MASK;
+    (*errorData).watchdogCausedLastReset =
+        data[0] & WATCHDOG_CAUSED_LAST_RESET_MASK;
+    (*errorData).configReadError =
+        data[0] & CONFIG_READ_ERROR_MASK;
+    (*errorData).rail15vUnderVoltageLockOut =
+        data[0] & RAIL_15V_UNDER_VOLTAGE_LOCK_OUT_MASK;
+    (*errorData).desaturationFault =
+        data[0] & DESATURATION_FAULT_MASK;
+
+    *rxErrorData = data[7];
+    *txErrorData = data[6];
+}
+
+float arrayToFloat(uint8_t* data)
+{
+    float result =
```

Comment:
Not that I'm opposed to old CCS code, but I didn't like having to do define an additional data structure. Also I can just copy paste from here so I'll do that :P